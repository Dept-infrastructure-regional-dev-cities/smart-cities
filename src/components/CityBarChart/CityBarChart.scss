@import '~style/variables';

.wrapper {
  padding: 16px;
}

.titleWrapper {
  display: flex;
  align-items: center;
  margin-bottom: 8px;
}

.title {
  @include type-intro-text;

  margin-right: 7px;
  line-height: 1;
}

.legendWrapper {
  display: flex;
  flex-flow: column wrap;
  justify-content: center;

  @include for-tablet-landscape-up {
    flex-direction: row;
  }
}
