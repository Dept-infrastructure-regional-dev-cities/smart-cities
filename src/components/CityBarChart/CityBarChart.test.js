import React from 'react';
import { shallow } from 'enzyme';
import CityBarChart from './CityBarChart';

const defaultProps = {
  colorBase: 'jobs',
  cities: [
    {
      name: 'Perth',
      indices: {
        population: 12,
        growthRate: 0.99,
      }
    },
    {
      name: 'Sydney',
      indices: {
        population: 12,
        growthRate: 0.19,
      }
    },
  ],
  indicators: [
    'population'
  ]
};

it('should match Snapshot', () => {
  const component = shallow(
    <CityBarChart {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});

describe('when displaying indicators with percentage units', () => {
  const component = shallow(
    <CityBarChart
      {...defaultProps}
      indicators={['growthRate']} // this has percentage units
      ceiling={1.1}
    />
  );

  const chartConfig = component.find('AbstractWidget').prop('config');

  it('should display labels with a "%" sign', () => {
    expect(chartConfig.yAxis.labels.format).toBe('{value}%');
  });

  it('should multiply values by 100', () => {
    expect(chartConfig.series[0].data).toEqual([99, 19]); // were 0.99 and 0.19
    expect(chartConfig.yAxis.ceiling).toBe(110);
  });
});

describe('when displaying indicators with numberic units', () => {
  const component = shallow(
    <CityBarChart
      {...defaultProps}
      indicators={['population']} // this has number units
    />
  );

  const chartConfig = component.find('AbstractWidget').prop('config');

  it('should not display a "%" sign for in labels', () => {
    expect(chartConfig.yAxis.labels.format).toBe('{value}');
  });


  it('should not multiply values by 100', () => {
    expect(chartConfig.series[0].data).toEqual([12, 12]);
  });
});

it('should render a legend when there is more than one indicator', () => {
  const component = shallow(
    <CityBarChart
      {...defaultProps}
      indicators={['population', 'growthRate']}
    />
  );

  expect(component.find('Legend').length).toBe(1);
});

it('should not render a legend when there is only one indicator', () => {
  const component = shallow(
    <CityBarChart
      {...defaultProps}
      indicators={['population']}
    />
  );

  expect(component.find('Legend').length).toBe(0);
});
