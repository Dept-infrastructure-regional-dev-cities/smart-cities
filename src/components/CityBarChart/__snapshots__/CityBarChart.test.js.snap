// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<div className=\\"wrapper\\">
  <div className=\\"titleWrapper\\">
    <h2 className=\\"title\\" style={{...}}>
      The population
    </h2>
    <AboutTooltip description=\\"TODO\\" position=\\"bottom\\" size={18} />
  </div>
  <AbstractWidget config={{...}} />
</div>"
`;
